spec/log_weasel/middleware_spec.rb

```diff
@@ -69,6 +69,53 @@
         end
       end
 
+      context "when the log weasel id is included in the params" do
+        let(:env) do
+          Rack::MockRequest.env_for("something", params: { logweasel_id: 'foo' })
+        end
+
+        context "with the environment variable enabled" do
+          it "sets LogWeasel::Transation.id to the header value" do
+            allow(ENV).to receive(:fetch).with('LOG_WEASEL_FROM_PARAMS', nil).and_return(true)
+
+            env['HTTP_X_REQUEST_ID'] = 'bar'
+
+            expect(StitchFix::LogWeasel::Transaction).to receive(:id=).with("bar")
+
+            StitchFix::LogWeasel::Middleware.new(app).call(env)
+          end
+
+          context "also includes a header in the request" do
```

Comment:
@brettfishman - the test case for when the param is present *AND* one of the headers is also present.  Good call out that this makes it a bit more explicit that its *intentional* that we would want this.

@rdadlani @almostwhitehat - I'm assuming this is ok for your purposes.  Basically the `TL;DR` is if a single request include one of the log weasel HTTP headers that would win out over the param value.  And in your specific use case, the browser won't pass any of those headers anyway so the ordering/precedence for your purposes is moot.